PythonPonies/bangazon-storefront
# Zgl tests

# Description
Add description about what this pull request covers.
- This PR covers the Order Model, Product On Order Model, product details, add products to order, and order views, as well as tests for creating and displaying and order
## Number of Fixes
Type of number of fixes included in this PR.

## Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
#1 
## Problem to Solve
Describe the problem(s) this pull request solves.
- Solves lack of orders and inability to add products to order
## Proposed Changes
Describe the proposed changes.
- adding Order, Product on Order models
- adding add product to order, display order and products views
- adding index, product detail, order templates
## Expected Behavior
Describe the expected behavior.
- if you create a superuser or login, should be able to add products to order and see them on order
## Steps to Test Solution

1. git fetch
2. remove database
3. migrate
4. makemigrations
5. createsuperuser
6. runserver
7. go to admin
8. create customer, product type, and product
9. go to url of product (usually localhost/1)
10. click 'button' at bottom of page
11. should take you to product types pages
12. if you look in database should see product and order under product on order
13. if you go to localhost/order should see product listed

## Testing

[X ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[X ] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
